### ISSUE TITLE
Traceback on MAC OS X Yosemite DevAssistant 0.10.1

### ISSUE THREAD

**Issue by syky27** [devassistant/devassistant#326]
```
╭─syky at aluminum in ~/tmp using ‹2.1.3›
╰─○ da create
No subassistants available.

To search DevAssistant Package Index (DAPI) for new assistants,
you can either browse https://dapi.devassistant.org/ or run

"da pkg search <term>".

Then you can run

"da pkg install <DAP-name>"

to install the desired DevAssistant package (DAP).
╭─syky at aluminum in ~/tmp using ‹2.1.3›
╰─○ da pkg search rails
ruby - Ruby assistants
╭─syky at aluminum in ~/tmp using ‹2.1.3›
╰─○ da pkg install rails
INFO: Installing DAP rails ...
ERROR: DAP rails not found.
╭─syky at aluminum in ~/tmp using ‹2.1.3›
╰─○ da pkg install ruby
INFO: Installing DAP ruby ...
INFO: Successfully installed DAPs ruby git common_args vim github
╭─syky at aluminum in ~/tmp using ‹2.1.3›
╰─○ da create ruby test
usage: create ruby [-h] {rails} ...
Action not found: "create ruby test"
╭─syky at aluminum in ~/tmp using ‹2.1.3›
╰─○ da create ruby
usage: create ruby [-h] {rails} ...
You have to select a subaction.
╭─syky at aluminum in ~/tmp using ‹2.1.3›
╰─○ da create ruby rails
usage: create ruby rails [-h] [-g [GITHUB]] -n NAME [-v]
create ruby rails: error: argument -n/--name is required
╭─syky at aluminum in ~/tmp using ‹2.1.3›
╰─○ da create ruby rails -n test
DevAssistant traceback (most recent call last):
Error: No DevAssistant frames to print.

Traceback (most recent call last):
  File "/usr/local/bin/da", line 9, in <module>
    load_entry_point('devassistant==0.10.1', 'console_scripts', 'da')()
  File "/Library/Python/2.7/site-packages/devassistant/cli/cli_runner.py", line 71, in run
    to_run.run(**parsed_args_decoded)
  File "/Library/Python/2.7/site-packages/devassistant/path_runner.py", line 52, in run
    self._run_path_dependencies(parsed_args)
  File "/Library/Python/2.7/site-packages/devassistant/path_runner.py", line 23, in _run_path_dependencies
    deps = self.path[-1].dependencies(parsed_args)
  File "/Library/Python/2.7/site-packages/devassistant/yaml_assistant.py", line 30, in inner
    return method(self, *args, **kwargs)
  File "/Library/Python/2.7/site-packages/devassistant/yaml_assistant.py", line 226, in dependencies
    deps.extend(lang.dependencies_section(sect, kwargs, runner=self))
  File "/Library/Python/2.7/site-packages/devassistant/lang.py", line 99, in dependencies_section
    deps.extend(Command(dep_type, dep_list, kwargs).run())
  File "/Library/Python/2.7/site-packages/devassistant/lang.py", line 54, in run
    for crs_prefix, crs in type(self).load_command_runners().command_runners.items():
  File "/Library/Python/2.7/site-packages/devassistant/lang.py", line 34, in load_command_runners
    cls.command_runners = utils.import_module('devassistant.command_runners')
  File "/Library/Python/2.7/site-packages/devassistant/utils.py", line 31, in import_module
    return importlib.import_module(module)
  File "/Library/Python/2.7/site-packages/devassistant/utils.py", line 22, in import_module
    fp, pathname, description = importlib.find_module(name.replace('.', '/'))
ImportError: No module named devassistant/command_runners
╭─syky at aluminum in ~/tmp using ‹2.1.3›
╰─○
```

**Comment by bkabrda** [comment]
I just tried fixing this in https://github.com/devassistant/devassistant/commit/8d29367cefe215c88d6d644dbb7e60d80ccfb2a0 - would you be able to test this on master branch @syky27 ?

(I still don't have a Mac, I just tried to explicitly provide lists of dirs to include from, which should theoretically make this work even if there's some sys.path weirdness. I'm not sure that's the case, but I have a hunch that it might be it...)

**Comment by hroncok** [comment]
I've been able to install Mac to VirtualBox. It is Mavericks, but I can try this.

**Comment by hroncok** [comment]
On master branch, still the error:

```
Miros-Mac-Pro:~ churchyard$ da crt ruby rails -n test
DevAssistant traceback (most recent call last):
Error: No DevAssistant frames to print.

Traceback (most recent call last):
  File "/usr/local/bin/da", line 9, in <module>
    load_entry_point('devassistant==0.10.1', 'console_scripts', 'da')()
  File "/Users/churchyard/Documents/devassistant/devassistant/cli/cli_runner.py", line 73, in run
    to_run.run(**parsed_args_decoded)
  File "/Users/churchyard/Documents/devassistant/devassistant/path_runner.py", line 52, in run
    self._run_path_dependencies(parsed_args)
  File "/Users/churchyard/Documents/devassistant/devassistant/path_runner.py", line 23, in _run_path_dependencies
    deps = self.path[-1].dependencies(parsed_args)
  File "/Users/churchyard/Documents/devassistant/devassistant/yaml_assistant.py", line 30, in inner
    return method(self, *args, **kwargs)
  File "/Users/churchyard/Documents/devassistant/devassistant/yaml_assistant.py", line 226, in dependencies
    deps.extend(lang.dependencies_section(sect, kwargs, runner=self))
  File "/Users/churchyard/Documents/devassistant/devassistant/lang.py", line 99, in dependencies_section
    deps.extend(Command(dep_type, dep_list, kwargs).run())
  File "/Users/churchyard/Documents/devassistant/devassistant/lang.py", line 54, in run
    for crs_prefix, crs in type(self).load_command_runners().command_runners.items():
  File "/Users/churchyard/Documents/devassistant/devassistant/lang.py", line 34, in load_command_runners
    cls.command_runners = utils.import_module('devassistant.command_runners')
  File "/Users/churchyard/Documents/devassistant/devassistant/utils.py", line 37, in import_module
    return importlib.import_module(module)
  File "/Users/churchyard/Documents/devassistant/devassistant/utils.py", line 28, in import_module
    dapath + sys.path)
ImportError: No module named devassistant/command_runners```

**Comment by hroncok** [comment]
This is what I read in the docs of `imp.find_module()`:

> This function does not handle hierarchical module names (names containing dots). In order to find P.M, that is, submodule M of package P, use find_module() and load_module() to find and load package P, and then use find_module() with the path argument set to P.__path__. When P itself has a dotted name, apply this recipe recursively.

While trying this on Python on Mac, it seems it works this way, will try to implement it, run test etc.

**Comment by hroncok** [comment]
Fixed.

Now of course, `da crt ruby rails -n test` fails on Mac using sed

**Comment by syky27** [comment]
You can use `sed` installed from `brew` just to run tests, than change tests so it will be able use BSD's sed

**Comment by hroncok** [comment]
Well, we use sed extensively in assistants (but not in tests), so we need to fix this in the assistants themselves. I'll look into that, after I fix our test suite, see #353
